src/components/oscal-utils/OSCALRestUtils.js

```diff
@@ -184,22 +186,51 @@ export function updateSspControlImplementationImplementedRequirementStatementByC
 ) {
   const partialRestImplementedRequirement = deepClone(implementedRequirement);
 
-  // Find our statementByComponent and update the description
-  const statementByComponent = partialRestImplementedRequirement.statements
-    .find((element) => element["statement-id"] === statementId)
-    ["by-components"].find(
-      (element) => element["component-uuid"] === componentId
-    );
+  // Find our statement
+  let statement = partialRestImplementedRequirement?.statements?.find(
+    (element) => element["statement-id"] === statementId
+  );
+
+  // If our statement doesn't exist, create and add it
+  if (!statement) {
+    statement = {
+      "statement-id": props.statementId,
+      uuid: uuidv4(),
+      "by-components": [],
+    };
+    if (!partialRestImplementedRequirement.statements) {
+      partialRestImplementedRequirement.statements = [];
+    }
+    partialRestImplementedRequirement.statements.push(statement);
+  }
+
+  // Find our statementByComponent
+  let statementByComponent = statement["by-components"].find(
+    (element) => element["component-uuid"] === componentId
+  );
+  // If our statementByComponent doesn't exit, create and add it
+  if (!statementByComponent) {
+    statementByComponent = {
+      "component-uuid": componentId,
+      uuid: uuidv4()
+    };
+    if (!statement["by-components"]) {
+      statement["by-components"] = [];
+    }
+    statement["by-components"].push(statementByComponent);
+  }
   statementByComponent.description = description;
 
   // Set each implementation parameter
-  if (implementationSetParameters) {
+  if (implementationSetParameters?.length > 0) {
     if (!statementByComponent["set-parameters"]) {
       statementByComponent["set-parameters"] = [];
     }
-    implementationSetParameters.forEach((element) =>
-      statementByComponent["set-parameters"].push(element)
-    );
+    implementationSetParameters.forEach((element) => {
+      if (element) {
+        statementByComponent["set-parameters"].push(element);
+      }
+    });
```

Comment:
GitHub won't let me give this as a typical suggestion because it covers deleted and unchanged lines. But from 225-233 should be able to be replaced with:

```ts
if (implementationSetParameters?.length) {
  statementByComponent["set-parameters"] ??= [];
  statementByComponent["set-parameters"].push(
    ...implementationSetParameters.filter((element) => !!element)
  );
}
```

Reply:
Addressed